src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/settings/LanguageServerConfigurable.java

```diff
@@ -0,0 +1,105 @@
+
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.ui.NamedConfigurable;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.util.NlsContexts;
+import com.redhat.devtools.intellij.quarkus.lsp4ij.LanguageServersRegistry;
+
+import javax.swing.*;
+
+/**
+ * UI settings to configure a given language server:
+ *
+ * <ul>
+ *     <li>Debug port</li>
+ *     <li>Suspend and wait for a debugger</li>
+ * </ul>
+ */
+public class LanguageServerConfigurable extends NamedConfigurable<LanguageServersRegistry.LanguageServerDefinition> {
+
+    private final LanguageServersRegistry.LanguageServerDefinition languageServerDefinition;
+
+    private LanguageServerView myView;
+
+    public LanguageServerConfigurable(LanguageServersRegistry.LanguageServerDefinition languageServerDefinition, Runnable updater) {
+        super(false, updater);
+        this.languageServerDefinition = languageServerDefinition;
+    }
+
+    @Override
+    public void setDisplayName(String name) {
+
+    }
+
+    @Override
+    public LanguageServersRegistry.LanguageServerDefinition getEditableObject() {
+        return languageServerDefinition;
+    }
+
+    @Override
+    public @NlsContexts.DetailedDescription String getBannerSlogan() {
+        return languageServerDefinition.getDisplayName();
+    }
+
+    @Override
+    public JComponent createOptionsPanel() {
+        if (myView == null) {
+            myView = new LanguageServerView(languageServerDefinition);
+        }
+        return myView.getComponent();
+    }
+
+    @Override
+    public @NlsContexts.ConfigurableName String getDisplayName() {
+        return languageServerDefinition.getDisplayName();
+    }
+
+    @Override
+    public boolean isModified() {
+        LanguageServerSettingsState.LanguageServerDefinitionSettings settings = LanguageServerSettingsState.getInstance()
+                .getLanguageServerSettings(languageServerDefinition.id);
+        if (settings == null) {
+            return true;
+        }
+        boolean modified = !myView.getDebugPort().equals(settings.getDebugPort());
+        modified = modified | myView.isDebugSuspend() != settings.isDebugSuspend();
+        return modified;
+    }
+
+    @Override
+    public void apply() throws ConfigurationException {
+        LanguageServerSettingsState.LanguageServerDefinitionSettings settings = new LanguageServerSettingsState.LanguageServerDefinitionSettings();
+        settings.setDebugPort(myView.getDebugPort());
+        settings.setDebugSuspend(myView.isDebugSuspend());
+        LanguageServerSettingsState.getInstance().setLanguageServerSettings(languageServerDefinition.id, settings);
+    }
+
+    @Override
+    public void reset() {
+        LanguageServerSettingsState.LanguageServerDefinitionSettings settings = LanguageServerSettingsState.getInstance()
+                .getLanguageServerSettings(languageServerDefinition.id);
+        if (settings != null) {

```

Comment:
shouldnt we at least log when no settings for a language server are found?

Reply:
Why you want to do that? I'm not sure it is relelvant to log that and it will polluate logs.

Reply:
ok, so if this is not unexpected, then it's ok